### ISSUE TITLE
Video descriptions don't parse YouTube redirect (shortened) urls entirely

### ISSUE THREAD

**Issue by ThisIsPaulDaily** [TeamNewPipe/NewPipe#5416]
### Checklist

- [x] I am using the latest version - 0.20.8 
- [ x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [ x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [ x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug

1. Go to a video with YouTube redirect URLs in the description that have been shortened by YouTube to obfuscate copy and pasting links. They do this to log outbound traffic clicks. 
2. Press on the hyperlink text that ends in an ellipses. Recognize that the link you arrived at is incomplete and the website doesn't load.
3. You can open the YouTube video in browser and click the link and you'll see the redirect link.

### Actual behaviour

For Example: https://www.youtube.com/watch?v=H97bzY8A9vI

Go to the description and note that several urls are cut off with an ellipsis (...) 

Click one such as this link from the description.

https://www.nytimes.com/2021/01/13/us
Note that the link is incomplete. 

### Expected behavior

On the website those links redirect through YouTube to log outbound traffic. 
It may look like this, but with a significantly longer ?redir_token that I have shortened for privacy and brevity.  
https://www.youtube.com/redirect?redir_token=QUFFBEI3dw%3D%3D&event=video_description&v=H97bzY8A9vI&q=https%3A%2F%2Fwww.nytimes.com%2F2021%2F01%2F13%2Fus%2Fpolitics%2Fairbnb-reservations-DC.html

In a perfect world, the app would parse links to remove the redirect link and bring it back to this. 

https://www.nytimes.com/2021/01/13/us/politics/airbnb-reservations-DC.html

### Device info

 - Android version/Custom ROM version:
Android 6.0.1 
 - Device model:
Kyocera Duraforce Pro

**Comment by eladkarako** [comment]
Do you mean something like 
<details><summary>this</summary>  

```js
(function(){"use strict";
  NodeList.prototype.forEach = Array.prototype.forEach;
  
  var query = 'a[href*="youtube.com/redirect"][href*="?q="],a[href*="youtube.com/redirect"][href*="&q="]'
     ,regex = /\/redirect.*[\?\&]q\=([^\&]+)/i
     ;

  document.querySelectorAll(query)
          .forEach(function(element){
            element.setAttribute("data-backup", element.href);
            try{
            element.href = self.decodeURIComponent( element.href.match(regex)[1] );
            }catch(err){}
          });
  /* also possible to also use 'https://unshorten.me/s/_______your_url______' for unshortening youtu.be, etc... */
}());
```
</details>
(just in Java..).

**Comment by TiA4f8R** [comment]
Fixed in 0.20.10
